libs/designsystem/src/lib/helpers/unique-id-generator.helper.ts

```diff
@@ -0,0 +1,19 @@
+export class UniqueIdGenerator {
+  private static registry = new Map<string, UniqueIdGenerator>();
+  private id = 0;
+
+  private constructor(private scope: string) {}
+
+  static scopedTo(scope: string) {
+    let scopedGenerator = UniqueIdGenerator.registry.get(scope);
+    if (!scopedGenerator) {
+      scopedGenerator = new UniqueIdGenerator(scope);
+      UniqueIdGenerator.registry.set(scope, scopedGenerator);
+    }
+    return scopedGenerator;
+  }
+
+  next(): string {
```

Comment:
next sounds like it could be dispatching an event like subject.next
```suggestion
  getUniqueId(): string {
```

Reply:
Ehhh not really :D
You have to see it in context - It's an id generator, so what do you think this does:
`idGenerator.next()`?